geode-core/src/test/java/org/apache/geode/cache/client/internal/PoolImplTest.java

```diff
@@ -137,6 +158,8 @@ private PoolImpl getPool(int retryAttemptsAttribute) {
     doReturn(1).when(poolAttributes).getMaxConnections();
     doReturn((long) 10e8).when(poolAttributes).getPingInterval();
     doReturn(retryAttemptsAttribute).when(poolAttributes).getRetryAttempts();
+    doReturn(requestLocatorInternalAddressEnabled).when(poolAttributes)
```

Comment:
The comment above this change states that these mocks are required for pool validation and setup to complete and that "The values of these attributes have no importance to the assertions of the test itself." In the case of this line, that's not true though, as the value of `requestLocatorInternalAddressEnabled` returned is explicitly what's being tested in the added test cases, and the pool can still be succesfully created with this line commented out. It might be better to change the signature of the `getPool()` method to take a `PoolAttributes` argument which can have the required additional mocking added in the individual test cases that require it, and leave `getPool()` to only do the mocking that's necessary to create a pool. I believe the same is true of the `retryAttemptsAttribute` argument, so that could also be removed from `getPool()` and the necessary mocking done only in the test methods that require it.